packages/neuron-wallet/src/block-sync-renderer/sync/light-connector.ts

```diff
@@ -0,0 +1,225 @@
+            scriptType: syncScript.scriptType,
+            cursor: undefined
+          })
+        }
+      })
+    }
+  }
+
+  private async subscribeSync() {
+    const minSyncBlockNumber = await SyncProgressService.getMinBlockNumber()
+    const header = await this.lightRpc.getTipHeader()
+    this.blockTipsSubject.next({
+      cacheTipNumber: minSyncBlockNumber,
+      indexerTipNumber: parseInt(header.number, 16)
+    })
+  }
+
+  private async initSyncProgress() {
+    if (!this.addressMetas.length) {
+      return
+    }
+    const sycnScripts = await this.lightRpc.getScripts()
+    const existSyncscripts: Record<string, LightScriptFilter> = {}
+    sycnScripts.forEach(v => {
+      existSyncscripts[scriptToHash(v.script)] = v
+    })
+    const allScripts = this.addressMetas
+      .map(addressMeta => {
+        const lockScripts = [
+          addressMeta.generateDefaultLockScript(),
+          addressMeta.generateACPLockScript(),
+          addressMeta.generateLegacyACPLockScript()
+        ]
+        return lockScripts.map(v => ({
+          script: v.toSDK(),
+          scriptType: 'lock' as CKBRPC.ScriptType,
+          walletId: addressMeta.walletId
+        }))
+      })
+      .flat()
+    const setScriptsParams = allScripts.map(v => ({
+      ...v,
+      blockNumber: existSyncscripts[scriptToHash(v.script)]?.blockNumber || '0x0'
+    }))
+    await this.lightRpc.setScripts(setScriptsParams)
+    await SyncProgressService.resetSyncProgress(allScripts)
+  }
+
+  private async initSync() {
+    await this.initSyncProgress()
+    while (this.pollingIndexer) {
+      await this.synchronize()
+      await scheduler.wait(5000)
+    }
+  }
+
+  private async syncNextWithScript({ script, scriptType, blockRange, cursor }: SyncQueueParam) {
+    const syncProgress = await SyncProgressService.getSyncStatus(script)
+    if (!syncProgress) {
+      return
+    }
+    const result = await this.lightRpc.getTransactions({ script, blockRange, scriptType }, 'asc', '0x64', cursor!)
+    if (!result.txs.length) {
+      await SyncProgressService.updateSyncStatus(syncProgress.hash, {
+        blockStartNumber: parseInt(blockRange[1]),
+        blockEndNumber: parseInt(blockRange[1]),
+        cursor: undefined
+      })
+      return
+    }
+    const scriptHash = scriptToHash(script)
+    this.transactionsSubject.next({ txHashes: result.txs.map(v => v.txHash), params: scriptHash })
+    this.syncInQueue.set(scriptToHash(script), {
+      blockStartNumber: result.lastCursor === '0x' ? parseInt(blockRange[1]) : parseInt(blockRange[0]),
+      blockEndNumber: parseInt(blockRange[1]),
+      cursor: result.lastCursor === '0x' ? undefined : result.lastCursor
+    })
+  }
+
+  private async collectLiveCellsByScript(query: LumosCellQuery) {
+    const { lock, type, data } = query
+    if (!lock && !type) {
+      throw new Error('at least one parameter is required')
```

Comment:
This error message is a bit obscure, how about `at least one script is required`